src/compiler/generators.ts

```diff
@@ -708,7 +708,7 @@ export function generateStructFieldMethods(
 
     const d = f.createGetAccessorDeclaration(
       [],
-      name,
+      name === "constructor" ? `$${name}` : name,
```

Comment:
In some `.capnp` files used in `workerd`, we have properties called `constructor`, which generated:
```ts
get constructor(): Constructor {
    $.utils.testWhich("constructor", $.utils.getUint16(0, this), 4, this);
    return $.utils.getStruct(0, Constructor, this);
}
```
which is invalid JS.
conversely, `capnp-ts` generates:
```
getConstructor() {
    capnp_ts_1.Struct.testWhich("constructor", capnp_ts_1.Struct.getUint16(0, this), 4, this);
    return capnp_ts_1.Struct.getStruct(0, Constructor, this);
}
```

I opted for changing the name of `constructor` properties to `$constructor` rather than reverting the `getter`/`setter` feature to reduce the diff of this PR.

Reply:
Ah interesting case. Looks good to me. My overall goal is to make generated interfaces seamless js-objects. Perhaps we can later fix it better using Proxy.